AppleEnrollment/BulkAssignProfile.ps1

```diff
@@ -0,0 +1,509 @@
+Test if the JSON is valid before calling the Graph REST interface
+.NOTES
+NAME: Test-JSON
+#>
+
+param (
+
+$JSON
+
+)
+
+    try {
+
+    $TestJSON = ConvertFrom-Json $JSON -ErrorAction Stop
+    $validJson = $true
+
+    }
+
+    catch {
+
+    $validJson = $false
+    $_.Exception
+
+    }
+
+    if (!$validJson){
+
+    Write-Host "Provided JSON isn't in valid JSON format" -f Red
+    break
+
+    }
+
+}
+ 
+####################################################
+
+Function Get-SkipToken(){
+
+<#
+.SYNOPSIS
+This function is used to get the skip token from given response string
+.DESCRIPTION
+The function tests get the skip token from given response string
+.EXAMPLE
+Get-SkipToken -Response $Response
+Get the skip token from given response
+.NOTES
+NAME: Get-SkipToken
+#>
+
+param (
+$response
+)
+
+    $skipToken = ""
+
+    if (![string]::IsNullOrWhiteSpace($response))
+    {
+        $token = "skiptoken="
+        [System.Int32]$index = $response.IndexOf($token)   
+        if ($index -gt 0)
+        {
+            [System.Int32]$nextLinkEndIndex = $response.IndexOf(';', $index + $token.Length)
+            $skipToken = $response.Substring($index + $token.Length, $nextLinkEndIndex - ($index + $token.Length))
+            $skipToken = [System.Web.HttpUtility]::UrlDecode($skipToken)
+        }
+    }
+
+    return $skipToken
+}
+
+####################################################
+
+Function Set-SkipToken(){
+
+<#
+.SYNOPSI
+This function is used to add the skip token to the given Uri
+.DESCRIPTION
+The function adds the skip token to the given Uri
+.EXAMPLE
+Set-SkipToken -RequestUri $requestUri, -Skiptoken $skipToken
+Set the skip token in the given Uri
+.NOTES
+NAME: Set-SkipToken
+#>
+
+param (
+[System.Uri]$requestUri,
+[System.String]$skipToken
+)
+    
+    [System.Uri]$newUri = [System.Uri]::new($requestUri)
+    [System.String]$skipTokenParameter = "`$skiptoken"
+
+    if (![System.String]::IsNullOrWhiteSpace($skipToken))
+    {
+        $qs = [System.Web.HttpUtility]::ParseQueryString($requestUri.Query)
+        $qs.Remove($skipTokenParameter)
+        $qs.Add($skipTokenParameter, $skipToken)
+
+        [System.UriBuilder] $uriBuilder = [System.UriBuilder]::new($requestUri)
+        $uriBuilder.Query = $qs.ToString()
+
+        $newUri = $uriBuilder.Uri
+    }
+
+    return $newUri;
+}
+
+####################################################
+
+Function Assign-ProfileToDevices(){
+<#
+.SYNOPSIS
+This function is used to assign a profile to given devices using the Graph API REST interface
+.DESCRIPTION
+The function connects to the Graph API Interface and assigns a profile to given devices
+.EXAMPLE
+Assign-ProfileToDevices
+Assigns a profile to given devices in Intune
+.NOTES
+NAME: Assign-ProfileToDevices
+#>
+
+[cmdletbinding()]
+
+param
+(
+    $Devices,
+    $ProfileId
+)
+
+$graphApiVersion = "Beta"
+$ResourceSegment = "deviceManagement/deviceManagement/enrollmentProfiles('{0}')/updateDeviceProfileAssignment"
```

Comment:
Should be `deviceManagement/enrollmentProfiles('{0}')/updateDeviceProfileAssignment`